### DISCUSSION TITLE
Classroom CLI, feature request: Cloning large number of student repos in parallel

### DISCUSSION THREAD

**Discussion by EtaoinWu** [community/community#124982] [GitHub Education] [answered]
### Hi everyone,

Currently `gh classroom clone student-repos` clones each student repositories in series. This results in painfully slow cloning (in a class of ~150 students, it took me ~15min to clone one homework; repo sizes average at 2MB). The repos are cloned in series, but since each repo is independent of others, would it be possible to add parallel mass-cloning as a quality of life feature?

I would suggest something like `-j <n>` to specify the number of parallel clone jobs.

**Comment by gitstua** [comment, marked as answer]
👋🏽  Not an official solution but you could use a script to clone the repos in parallel if you can get the list into a file

```
#!/bin/bash

# Name of the file containing the repo URLs
REPO_LIST="repos.txt"

# Function that clones a repo
clone_repo() {
    git clone "$1" &
}

# Read the repo list file line by line
while IFS= read -r line
do
    # Clone the repo in the background
    clone_repo "$line"
done < "$REPO_LIST"

# Wait for all background jobs to finish
wait
```
